lib/kubernetes/src/equality/utils.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2020 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// empty lists can either be `undefined` or have length of 0
+const isResourceListEmpty = <Resource>(list: Array<Resource> | void) =>
+  !list || list.length === 0;
+
+export const isResourceListEqual = <Resource>(
+  desiredList: Array<Resource> | void,
+  existingList: Array<Resource> | void,
+  isResourceEqual: (desired: Resource, existing: Resource) => boolean
+): boolean => {
+  if (!Array.isArray(desiredList) || !Array.isArray(existingList)) {
+    return (
+      isResourceListEmpty(desiredList) && isResourceListEmpty(existingList)
+    );
+  }
```

Comment:
My guess is that this piece of code might not be clear enough - please let me know if I need to improve the expressiveness!

Reply:
When you write code that doesn't look clear to you, always comment on it. If you feel you can improve it, please do otherwise leave a comment for the next person that reads this code.